web/chromecom.js

```diff
@@ -308,16 +308,36 @@ class ChromePreferences extends BasePreferences {
         // Get preferences as set by the system administrator.
         // See extensions/chromium/preferences_schema.json for more information.
         // These preferences can be overridden by the user.
-        chrome.storage.managed.get(this.defaults, function(items) {
+
+        // Deprecated preferences are removed from web/default_preferences.json,
+        // but kept in extensions/chromium/preferences_schema.json for backwards
+        // compatibility with managed preferences.
+        let defaultManagedPrefs = Object.assign({
+          enableHandToolOnLoad: false,
+          disableTextLayer: false,
+          enhanceTextSelection: false,
+        }, this.defaults);
+
+        chrome.storage.managed.get(defaultManagedPrefs, function(items) {
+          items = items || defaultManagedPrefs;
           // Migration code for https://github.com/mozilla/pdf.js/pull/7635.
           // Never remove this, because we have no means of modifying managed
           // preferences.
-          if (items && items.enableHandToolOnLoad && !items.cursorToolOnLoad) {
+          if (items.enableHandToolOnLoad && !items.cursorToolOnLoad) {
             // if the old enableHandToolOnLoad has a non-default value,
             // and cursorToolOnLoad has a default value, migrate.
-            items.enableHandToolOnLoad = false;
             items.cursorToolOnLoad = 1;
           }
+          delete items.enableHandToolOnLoad;
+          if (items.textLayerMode !== 1) {
```

Comment:
Shouldn't the condition be `items.textLayerMode === 1` here, given that otherwise a non-default `textLayerMode` (as set by the user) will always be overwritten which seem strange!?

Reply:
The code here is only concerned with reading and parsing managed preferences. When `getPreferences(items)` is called, the given items specify the values to retrieve, along with the default values if no explicit value is set by the user. Lines 308 - 310 briefly state the purpose of the code (without explicitly stating the implementation details that I've written here).